osu.Game/Overlays/Mods/ModSelectScreen.cs

```diff
@@ -31,16 +33,43 @@ public class ModSelectScreen : OsuFocusedOverlayContainer
 
         protected override bool StartHidden => true;
 
+        private Func<Mod, bool> isValidMod = m => true;
+
+        public Func<Mod, bool> IsValidMod
+        {
+            get => isValidMod;
+            set
+            {
+                isValidMod = value ?? throw new ArgumentNullException(nameof(value));
+
+                if (IsLoaded)
+                    updateAvailableMods();
+            }
+        }
+
+        /// <summary>
+        /// Whether configurable <see cref="Mod"/>s can be configured by the local user.
+        /// </summary>
+        protected virtual bool AllowConfiguration => true;
+
+        /// <summary>
+        /// Whether the total score multiplier calculated from the current selected set of mods should be shown.
+        /// </summary>
+        protected virtual bool ShowTotalMultiplier => true;
```

Comment:
Not sure about these kinds of properties still. Where they should likely be in the constructor. But I guess it's fine since we understand the intention of them?

Reply:
I'm fine with moving to ctor. The only reason why this is done so is because the old overlay did the same.

Reply:
Note that I'm still not sure on this. And when "too much" is in the constructor (ie. we should make a construction parameters class and add more overhead).

Fine to leave for now, but interested in more opinions.